message Command_Ping {
}

message Command_Login {
	optional string user_name = 1;
	optional string password = 2;
}

message Command_Message {
	optional string user_name = 1;
	optional string message = 2;
}

message Command_ListUsers {
}

message Command_GetGamesOfUser {
	optional string user_name = 1;
}

message Command_GetUserInfo {
	optional string user_name = 1;
}

message Command_AddToList {
	optional string list = 1;
	optional string user_name = 2;
}

message Command_RemoveFromList {
	optional string list = 1;
	optional string user_name = 2;
}

message Command_DeckList {
}

message Command_DeckNewDir {
	optional string path = 1;
	optional string dir_name = 2;
}

message Command_DeckDelDir {
	optional string path = 1;
}

message Command_DeckDel {
	optional uint32 deck_id = 1;
}

message Command_DeckDownload {
	optional uint32 deck_id = 1;
}

message Command_DeckUpload {
	optional string path = 1;
	optional string deck_list = 2;
}

message Command_ListRooms {
}

message Command_JoinRoom {
	optional uint32 room_id = 1;
}

message SessionCommand {
	enum SessionCommandType {
		PING = 1000;
		LOGIN = 1001;
		MESSAGE = 1002;
		LIST_USERS = 1003;
		GET_GAMES_OF_USER = 1004;
		GET_USER_INFO = 1005;
		ADD_TO_LIST = 1006;
		REMOVE_FROM_LIST = 1007;
		DECK_LIST = 1008;
		DECK_NEW_DIR = 1009;
		DECK_DEL_DIR = 1010;
		DECK_DEL = 1011;
		DECK_DOWNLOAD = 1012;
		DECK_UPLOAD = 1013;
		LIST_ROOMS = 1014;
		JOIN_ROOM = 1015;
	}
	extensions 100 to max;
}

extend SessionCommand {
	optional Command_Ping command_ping = 1000;
	optional Command_Login command_login = 1001;
	optional Command_Message command_message = 1002;
	optional Command_ListUsers command_list_users = 1003;
	optional Command_GetGamesOfUser command_get_games_of_user = 1004;
	optional Command_GetUserInfo command_get_user_info = 1005;
	optional Command_AddToList command_add_to_list = 1006;
	optional Command_RemoveFromList command_remove_from_list = 1007;
	optional Command_DeckList command_deck_list = 1008;
	optional Command_DeckNewDir command_deck_new_dir = 1009;
	optional Command_DeckDelDir command_deck_del_dir = 1010;
	optional Command_DeckDel command_deck_del = 1011;
	optional Command_DeckDownload command_deck_download = 1012;
	optional Command_DeckUpload command_deck_upload = 1013;
	optional Command_ListRooms command_list_rooms = 1014;
	optional Command_JoinRoom command_join_room = 1015;
}
 
